packages/malloy/src/lang/field-space.ts

```diff
@@ -243,30 +260,34 @@ export class NewFieldSpace extends StructSpace {
       // one of three kinds of fields are legal in an explore: expressions ...
       const elseLog = def.log;
       const elseType = def.elementType;
-      if (def instanceof ExprFieldDecl) {
+      if (def instanceof FieldDeclaration) {
         const exprField = new ExpressionFieldFromAst(this, def);
         this.setEntry(exprField.name, exprField);
         // querry (turtle) fields
       } else if (def instanceof TurtleDecl) {
         const name = def.name;
         this.setEntry(name, new QueryFieldAST(this, def, name));
       } else if (def instanceof RenameField) {
-        if (def.oldName === def.newName) {
+        if (def.oldName.refString === def.newName) {
           def.log("Can't rename field to itself");
           continue;
         }
-        const oldValue = this.entry(def.oldName);
-        if (oldValue instanceof SpaceField) {
-          this.setEntry(
-            def.newName,
-            new RenameSpaceField(oldValue, def.newName, def.location)
-          );
-          this.dropEntry(def.oldName);
+        const oldValue = def.oldName.getField(this);
```

Comment:
I still wonder if this should count as a reference

Reply:
`file1.malloy`
```
explore: base is table('somewhere') {
  dimension: thing_with_horrible_name is COMPLEX_EXPRESSION;
}
```
`file2.malloy`
```
import "file1.malloy"

explore: thing is base {
  rename: thing is thing_with_horrible_name
}
```

In this case, I want to Command+Click on `thing_with_horrible_name` and see how it's defined.